src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs

```diff
@@ -820,32 +820,27 @@ private CodeGenOutput GenerateFastifyPluginOutput(ServiceInfo service)
 
 			return new CodeGenOutput(file);
 
-			string GetJsonSchemaType(string typeName)
+			string GetJsonSchemaType(ServiceTypeInfo? serviceType)
 			{
-				if (typeName.EndsWith("[]", StringComparison.Ordinal))
-					return $"{{ type: 'array', items: {GetJsonSchemaType(typeName.Substring(0, typeName.Length - 2))} }}";
-				if (typeName.StartsWith("nullable<", StringComparison.Ordinal))
-					return $"{{ oneOf: [ {GetJsonSchemaType(typeName.Substring(9, typeName.Length - 10))}, {{ type: 'null' }} ] }}";
-				if (typeName.StartsWith("result<", StringComparison.Ordinal))
-					return $"{{ type: 'object', properties: {{ value: {GetJsonSchemaType(typeName.Substring(7, typeName.Length - 8))}, error: {{ $ref: '_error' }} }} }}";
-				if (typeName.StartsWith("map<", StringComparison.Ordinal))
-					return $"{{ type: 'object', additionalProperties: {GetJsonSchemaType(typeName.Substring(4, typeName.Length - 5))} }}";
-				if (typeName is "error")
-					return $"{{ $ref: '{"_error"}' }}";
-				if (customTypes.Contains(typeName))
-					return $"{{ $ref: '{typeName}' }}";
-				if (typeName is "string" or "bytes" or "datetime")
-					return $"{{ type: '{"string"}' }}";
-				if (typeName is "boolean")
-					return $"{{ type: '{"boolean"}' }}";
-				if (typeName is "int32" or "int64")
-					return $"{{ type: '{"integer"}' }}";
-				if (typeName is "decimal" or "double")
-					return $"{{ type: '{"number"}' }}";
-				if (typeName is "object")
-					return $"{{ type: '{"object"}', additionalProperties: true }}";
-
-				throw new NotSupportedException("Unknown type " + typeName);
+				if (serviceType is null)
+					throw new ArgumentNullException(nameof(serviceType));
+
+				return serviceType.Kind switch
+				{
+					ServiceTypeKind.String or ServiceTypeKind.Bytes or ServiceTypeKind.DateTime or ServiceTypeKind.ExternalEnum => $"{{ type: '{"string"}' }}",
```

Comment:
```suggestion
					ServiceTypeKind.String or ServiceTypeKind.Bytes or ServiceTypeKind.DateTime or ServiceTypeKind.ExternalEnum => "{ type: 'string' }",
```

Reply:
So I did have this originally, but was getting a `Probable JSON string detected (JSON002)` linting error, which seemed strange. I'm not actually sure where it is coming from, I can't find anything online about it. I can disable it inline for this method, or maybe make it a suggestion instead of a an error?

Reply:
Here's what I've found on that now: https://www.infoq.com/news/2022/11/visual-studio-tagged-strings/

Basically, I can add `/*lang=json*/` to each line to let Roslyn check it for syntax errors. TIL.

Reply:
I think the comments hurt readability though, so I vote for adding `dotnet_diagnostic.JSON002.severity = suggestion` to `.editorconfig`.

Reply:
fix: 580c68bdc525f194adfa7923f1759673899d52a1

Reply:
TIL